Gems/Atom/Feature/Common/Code/Source/SkyAtmosphere/SkyAtmosphereFeatureProcessor.cpp

```diff
@@ -0,0 +1,355 @@
+        m_passNeedsUpdate = true;
+    }
+
+    void SkyAtmosphereFeatureProcessor::SetSunRadiusFactor(AtmosphereId id, float factor)
+    {
+        auto& params = m_params.GetElement(id.GetIndex());
+        params.m_sunRadiusFactor = factor;
+        m_passNeedsUpdate = true;
+    }
+
+    void SkyAtmosphereFeatureProcessor::InitializeAtmosphere(AtmosphereId id)
+    {
+        m_passNeedsUpdate = true;
+        for (auto pass : m_skyAtmosphereParentPasses )
+        {
+            pass->CreateAtmospherePass(id);
+        }
+    }
+
+    void SkyAtmosphereFeatureProcessor::OnRenderPipelinePassesChanged([[maybe_unused]]RPI::RenderPipeline* renderPipeline)
+    {
+        CachePasses();
+        UpdateBackgroundClearColor();
+    }
+
+    void SkyAtmosphereFeatureProcessor::OnRenderPipelineAdded([[maybe_unused]]RPI::RenderPipelinePtr renderPipeline)
+    {
+        CachePasses();
+        UpdateBackgroundClearColor();
+    }
+
+    void SkyAtmosphereFeatureProcessor::OnRenderPipelineRemoved([[maybe_unused]] RPI::RenderPipeline* renderPipeline)
+    {
+        CachePasses();
+    }
+    
+    void SkyAtmosphereFeatureProcessor::CachePasses()
+    {
+        m_skyAtmosphereParentPasses.clear();
+
+        RPI::PassFilter passFilter = RPI::PassFilter::CreateWithTemplateName(Name("SkyAtmosphereParentTemplate"), GetParentScene());
+        RPI::PassSystemInterface::Get()->ForEachPass(passFilter, [this](RPI::Pass* pass) -> RPI::PassFilterExecutionFlow
+            {
+                SkyAtmosphereParentPass* parentPass = static_cast<SkyAtmosphereParentPass*>(pass);
+                m_skyAtmosphereParentPasses.emplace_back(parentPass);
+                return RPI::PassFilterExecutionFlow::ContinueVisitingPasses;
+            });
+
+
+        // make sure we update all the LUTS
+        for (auto id : m_atmosphereIds)
+        {
+            if (id.IsValid())
+            {
+                auto& params = m_params.GetElement(id.GetIndex());
+                params.m_lutUpdateRequired = true;
+
+                // make sure all removed atmosphere passes are restored
+                InitializeAtmosphere(id);
+            }
+        }
+
+        m_passNeedsUpdate = true;
+    }
+    
+    void SkyAtmosphereFeatureProcessor::Simulate([[maybe_unused]] const FeatureProcessor::SimulatePacket& packet)
+    {
+        AZ_PROFILE_SCOPE(RPI, "SkyAtmosphereFeatureProcessor: Simulate");
+
+        if (m_passNeedsUpdate)
+        {
+            for (auto pass : m_skyAtmosphereParentPasses)
+            {
+                for (auto id : m_atmosphereIds)
+                {
+                    if (id.IsValid())
+                    {
+                        auto& params = m_params.GetElement(id.GetIndex());
+                        pass->UpdateAtmospherePassSRG(id, params);
+
+                        // reset the lut update flag now that we've passed on the changes
+                        params.m_lutUpdateRequired = false;
```

Comment:
nit: It feels like when the update flag is 'dirty' you can simply update all parameters and compile the Srg rather than gather one by one and set selected - KISS. It really doesn't save much since you compile and copy the srg anyway as a result.

Reply:
yup, it's really just there to make sure the transmittance LUT pass is enabled so that LUT gets re-rendered.  Then it is immediately turned off until those LUT params change again and the LUT needs to be updated.